src/Compilers/VisualBasic/Portable/Compilation/VisualBasicDeterministicKeyBuilder.vb

```diff
@@ -0,0 +1,94 @@
+﻿' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+
+Imports System.Globalization
+
+Namespace Microsoft.CodeAnalysis.VisualBasic
+
+    Friend NotInheritable Class VisualBasicDeterministicKeyBuilder
+        Inherits DeterministicKeyBuilder
+
+        Friend Shared ReadOnly Instance As New VisualBasicDeterministicKeyBuilder()
+
+        Private Sub New()
+
+        End Sub
+
+        Protected Overrides Sub WriteParseOptionsCore(writer As JsonWriter, parseOptions As ParseOptions)
+            ' This can happen for SyntaxTree that are constructed via the API. 
+            If parseOptions Is Nothing Then
+                Return
+            End If
+
+            Dim basicOptions = TryCast(parseOptions, VisualBasicParseOptions)
+            If basicOptions Is Nothing Then
+                Throw New InvalidOperationException()
+            End If
+
+            MyBase.WriteParseOptionsCore(writer, parseOptions)
+
+            writer.Write("languageVersion", basicOptions.LanguageVersion)
+            writer.Write("specifiedLanguageVersion", basicOptions.SpecifiedLanguageVersion)
+
+            writer.WriteKey("preprocessorSymbols")
+            writer.WriteObjectStart()
+            For Each pair In basicOptions.PreprocessorSymbols.OrderBy(Function(x, y) StringComparer.Ordinal.Compare(x.Key, y.Key))
+                Dim value = pair.Value
+                If value Is Nothing Then
+                    writer.WriteNull(pair.Key)
+                    Continue For
+                End If
+
+                writer.WriteKey(pair.Key)
+
+                Dim type = value.GetType()
+                If type = GetType(String) Then
```

Comment:
Do we have test coverage for all cases here? I saw tests with string and with integer values, but not boolean.

Reply:
Yep was missing `bool` and `float`. fixed.